eclair-core/src/test/scala/fr/acinq/eclair/UInt64Spec.scala

```diff
@@ -31,6 +31,7 @@ class UInt64Spec extends FunSuite {
     assert(b < a)
     assert(z < a && z < b && z < c)
     assert(a == a)
+    assert(a == UInt64.MaxValue)
```

Comment:
I think it's also worth enriching these tests with more edge cases around `Long.MaxValue`.
I'd also test the toString method.

Reply:
Overall this `UInt64` is full of subtleties and everything could go wrong if we didn't handle an edge case well, so I'd add a lot more things here to feel safe...
That's also a good candidate for fuzzing when we add that to eclair.

Reply:
I did some similar testing in [here](https://github.com/ACINQ/eclair/blob/master/eclair-core/src/test/scala/fr/acinq/eclair/CoinUtilsSpec.scala#L32), it's related but i think is worth repeating the tests in `UInt64Spec`. As you said this is plenty of subtleties and thorough testing is the first step to get them right.